test/runnable/casting.d

```diff
@@ -199,6 +199,34 @@ void test11722()
 }
 
 /***************************************************/
+// 14218
+
+void test14218()
+{
+    foreach (To; Seq!( byte,  short,  int,  long,
+                      ubyte, ushort, uint, ulong,
+                       char,  wchar, dchar, bool))
+    {
+        auto x = cast(To)null;
+        assert(x == 0);     // false, '0x00'
+    }
+
+    // Questionable but currently accepted
+    foreach (To; Seq!( float,  double,  real,
+                      ifloat, idouble, ireal))
+    {
+        auto x = cast(To)null;
+        assert(x == 0);     // 0i
+    }
```

Comment:
With gcc/g++/dmc, casting void pointer to any floating point types are not allowed.
I'm not sure why the cast is accepted in D, so I said it's "questionable".